apps/picsa-tools/resources-tool/src/app/components/resource-item/file/file.html

```diff
@@ -26,8 +26,13 @@
     </div>
   </div>
 
+  <!-- Add thumbnail status text here, it is only undefined when loading, on failure it would be an empty string  -->
+  @if (thumbnail === undefined && resource.subtype === 'video' && !attachment) {
+  <div  class="thumbnail-status">loading...</div>
+  }
```

Comment:
nit(blocking): I think it would be good to add a component here that can manage the thumbnail, receiving the URL as an input e.g. `<picsa-video-thumbnail [url]="dbDoc._data.url">`

You can also move the logic and styling so that it is easier to adapt and maintain in the future

Reply:
Alright